.circleci/config.yml

```diff
@@ -1,19 +1,91 @@
+e2eTest: &e2eTest
+    machine: true
+    working_directory: /home/circleci/.go_workspace/src/github.com/giantswarm/cert-exporter
+    steps:
+    - checkout
+
+    - attach_workspace:
+        at: .
+
+    - run:
+        name: Get e2e-harness binary
+        command: |
+          wget -q $(curl -sS https://api.github.com/repos/giantswarm/e2e-harness/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
+          chmod +x ./e2e-harness
+
+    - run: ./e2e-harness localkube
+
+    - run: ./e2e-harness setup --remote=false
+    
+    - run:
+        name: Publish chart to CNR using a temporary channel for testing
+        command: ./architect publish --pipeline=false --channels=${CIRCLE_SHA1}-${TEST_NAME}
+
+    - run: ./e2e-harness test --test-dir=${TEST_DIR}
+
+    - run:
+        name: Finish with cleanup, no matter if the test succeeded or not
+        command: |
+          ./architect unpublish --channels=${CIRCLE_SHA1}-${TEST_NAME}
+          ./e2e-harness teardown
+        when: always
+
 version: 2
 jobs:
   build:
     machine: true
     steps:
     - checkout
 
-    - run: |
-        wget -q $(curl -sS -H "Authorization: token $RELEASE_TOKEN" https://api.github.com/repos/giantswarm/architect/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
-        chmod +x ./architect
-        ./architect version
+    - run:
+        name: Get architect binary
+        command: |
+          wget -q $(curl -sS -H "Authorization: token $RELEASE_TOKEN" https://api.github.com/repos/giantswarm/architect/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
+          chmod +x ./architect
+          ./architect version
+
     - run: ./architect build
     - store_test_results:
         path: /tmp/results
-    - deploy:
-        command: |
-          if [ "${CIRCLE_BRANCH}" == "master" ]; then
-            ./architect deploy
-          fi
+
+    - persist_to_workspace:
+        root: .
+        paths:
+        - ./architect
+
+  e2eTestBasic:
+    environment:
+      TEST_DIR: "integration/test/basic"
+      TEST_NAME: "basic"
+    <<: *e2eTest
+
+  deploy:
+    machine: true
+    steps:
+    - checkout
+
+    - attach_workspace:
+        at: .
+
+    - run: ./architect deploy
+
+    - run: |
+        sed -i 's/version:.*/version: 1.0.0-'"${CIRCLE_SHA1}"'/' helm/cert-exporter-chart/Chart.yaml
```

Comment:
This sed magic is needed to support deploying with draughtsman. See next comment for the Chart.yaml change.